Jenkinsfile

```diff
@@ -78,13 +79,19 @@ def cmake_build(Map conf=[:]){
         setup_args = " -DCMAKE_CXX_COMPILER_LAUNCHER='ccache' -DCMAKE_C_COMPILER_LAUNCHER='ccache' " + setup_args
     }
 
+    if ( build_fin == "ON" )
+    {
+        setup_args = " -DMIOPEN_INSTALL_CXX_HEADERS=On " + setup_args
+    }
+
```

Comment:
Can we move FIN-related stuff to fin directory?

Reply:
I modified the build order here because using MIOpen's CMakeList to build Fin was causing conflicts. This should be here so that we can use Fin in the MIOpen CI. This has the effect it had before of ensuring Fin is compatible with MIOpen and enables features like the perf db validity check.

Reply:
It is not necessary to use MIOpens CMakeList to build Fin, you can use CMakeList from fin directory. Also I googled a bit and it's possible to include one Jenkinsfile to another. You can store Fin Jenkinsfile in fin directory and import from it anything you need: functions, variables and even whole stages.

Reply:
@averinevg here has been altered so that the cmakes are no longer intertwined.
As for loading the fin jenkinsfile, I don't think that will work. The load functionality in Jenkinsfile appears to work well for groovy files, but because a pipeline{...} definition is present in fin/Jenkinsfile, attempts to load the file will cause an error, as multiple definitions of these are not allowed.
If it would be sufficient to move the fin related build steps into a separate cmake_build function, I can provide that.